tests/test_models.py

```diff
@@ -0,0 +1,1001 @@
+
+    prob_check = [1.0, 0.9999945163726807]
+    pred_check = [5, 8]
+    for idx, probs_ in enumerate(probs):
+        prob_max = max(probs_)
+        assert (
+            np.abs(prob_max - prob_check[idx]) <= 1e-8 and preds[idx] == pred_check[idx]
+        )
+
+
+def test_patch_predictor_mobilenet_v3_small_kather100K():
+    """Test for patch predictor with mobilenet_v3_small on Kather 100K dataset."""
+    file_parent_dir = pathlib.Path(__file__).parent
+    dir_patches = file_parent_dir.joinpath("data/sample_patches/")
+    list_paths = grab_files_from_dir(dir_patches, file_types="*.tif")
+    dataset = Patch_Dataset(list_paths)
+
+    # API 1, also test with return_labels
+    predictor = CNN_Patch_Predictor(
+        predefined_model="mobilenet_v3_small-kather100K", batch_size=1
+    )
+    # don't run test on GPU
+    output = predictor.predict(
+        dataset, return_probs=True, return_labels=True, on_gpu=False
+    )
+    probs = output["probs"]
+    preds = output["preds"]
+    labels = output["labels"]
+
+    assert len(probs) == len(preds)
+    assert len(probs) == len(labels)
+
+    prob_check = [1.0, 0.9999963045120239]
+    pred_check = [5, 8]
+    for idx, probs_ in enumerate(probs):
+        prob_max = max(probs_)
+        assert (
+            np.abs(prob_max - prob_check[idx]) <= 1e-8 and preds[idx] == pred_check[idx]
+        )
+
+
+# -------------------------------------------------------------------------------------
+# Command Line Interface
+# -------------------------------------------------------------------------------------
+
+
+def test_command_line_patch_predictor():
+    """Test for the patch predictor CLI."""
+    file_parent_dir = pathlib.Path(__file__).parent
+    dir_patches = file_parent_dir.joinpath("data/sample_patches/")
+
+    runner = CliRunner()
+    patch_predictor_dir = runner.invoke(
+        cli.main,
+        [
+            "patch-predictor",
+            "--predefined_model",
+            "resnet18-kather100K",
+            "--img_input",
+            dir_patches,
+            "--batch_size",
+            2,
+            "--return_probs",
+            False,
+        ],
+    )
+
+    assert patch_predictor_dir.exit_code == 0
+
+    list_paths = grab_files_from_dir(dir_patches, file_types="*.tif")
+    single_path = list_paths[0]
+    patch_predictor_single_path = runner.invoke(
+        cli.main,
+        [
+            "patch-predictor",
+            "--predefined_model",
+            "resnet18-kather100K",
+            "--img_input",
+            single_path,
+            "--batch_size",
+            2,
+            "--return_probs",
+            False,
+        ],
+    )
+
+    assert patch_predictor_single_path.exit_code == 0
+
+
+def test_command_line_patch_predictor_crash():
+    """Test for the patch predictor CLI."""
+    file_parent_dir = pathlib.Path(__file__).parent
+    img_path = file_parent_dir.joinpath("data/sample_patches/kather1_unknown.tif")
```

Comment:
please use tiatoolbox server for this.

Reply:
Addressed in [this commit ](https://github.com/TIA-Lab/tiatoolbox/pull/98/commits/6fa8ac4775a3f63b6b9f522454967902d54b2d68)